mycroft/listener/__init__.py

```diff
@@ -176,9 +185,18 @@ def process(self, audio, lang=None):
                 transcription = self.transcribe(audio, lang)
             if transcription:
                 ident = str(stopwatch.timestamp) + str(hash(transcription))
+
+                # clean wake words caught at start of transcript
+                clean = transcription
+                if self.listening_mode == ListeningMode.CONTINUOUS:
+                    for ww in self._listen_strings:
+                        if clean.startswith(ww):
```

Comment:
We might want to check for full words (`clean.startswith(f"{ww} ")`) in case the ww is a substring, i.e.: ww="alexa"  transcription="alexander"

Reply:
I think a text/audio transformer makes sense; we can just use message context to determine if a WW was recognized.

In the case of audio plugin, pass the audio through some theoretical ww recognizer that gives us timestamps to trim the ww audio.

For a text plugin, use whatever logic to determine if the ww should be trimmed from the start/end and maybe use something like `voc_match` to handle homonyms in transcriptions